src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -9188,10 +9212,33 @@ public void M(in int x)
 }
 ";
 
-            var comp = CompileAndVerify(source);
+            var source_skip = @"
+[module: System.Runtime.CompilerServices.SkipLocalsInitAttribute]
+
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+
+public class C
+{
+    public void M(in int x)
+    {
+    }
+}
+";
+
+
+            var comp_init = CompileAndVerify(source_init);
+            var comp_skip = CompileAndVerify(source_skip);
 
-            Assert.Null(comp.HasLocalsInit("Microsoft.CodeAnalysis.EmbeddedAttribute..ctor"));
-            Assert.Null(comp.HasLocalsInit("System.Runtime.CompilerServices.IsReadOnlyAttribute..ctor"));
+            Assert.Null(comp_init.HasLocalsInit("Microsoft.CodeAnalysis.EmbeddedAttribute..ctor"));
+            Assert.Null(comp_init.HasLocalsInit("System.Runtime.CompilerServices.IsReadOnlyAttribute..ctor"));
+
+            Assert.Null(comp_skip.HasLocalsInit("Microsoft.CodeAnalysis.EmbeddedAttribute..ctor"));
+            Assert.Null(comp_skip.HasLocalsInit("System.Runtime.CompilerServices.IsReadOnlyAttribute..ctor"));
```

Comment:
It looks like the difference is not tested currently. Perhaps inspect the synthesized methods directly:
```
CompileAndVerify(source_init, symbolValidator: module =>
    {
        var type = module.ContainingAssembly.GetTypeByMetadataName("Microsoft.CodeAnalysis.EmbeddedAttribute");
        Assert.True(type.Constructors[0].AreLocalsZeroed);
    });
```

Reply:
When we call type.Constructors[0], we get a `PEMethodSymbol` in place of a `SynthesizedEmbeddedAttributeConstructorSymbol`. `PEMethodSymbol` currently throws an unreachable exception for AreLocalsZeroed property.

Reply:
Sorry, the lambda should be the `sourceSymbolValidator` argument, not `symbolValidator`.

Reply:
This won't work either. `GetTypeByMetadataName` returns null. I debugged through it and basically the type can't be found because there's no `Microsoft` namespace under global namespace in source (which makes sense).